pdns/dnsdist-web.cc

```diff
@@ -386,6 +386,127 @@ static void connectionThread(int sock, ComboAddress remote, string password, str
         resp.status=404;
       }
     }
+    else if (req.url.path == "/metrics") {
+        handleCORS(req, resp);
+        resp.status = 200;
+
+        std::ostringstream output;
+        for (const auto& e : g_stats.entries) {
+          std::string metricName = std::get<0>(e);
+
+          // Prometheus suggest using '_' instead of '-'
+          std::string prometheusMetricName = "dnsdist_main_" + boost::replace_all_copy(metricName, "-", "_");
```

Comment:
I think we only added "main" in the carbon output because metronome would not ingest it otherwise. If so (cc @ahupowerdns) I suggest we don't add the "main" to the prometheus output.

Reply:
Let is not continue that hack indeed. Sorry :-)

Reply:
I removed "main" from all metric names and added up to date raw output for /metrics.